Code/Framework/AzCore/AzCore/Math/Uuid.cpp

```diff
@@ -245,199 +31,57 @@ namespace AZ
+#if AZ_TRAIT_UUID_SUPPORTS_GUID_CONVERSION
+    Uuid::Uuid(const GUID& guid)
     {
-        unsigned char val = data[6];
-        if ((val & 0xF0) == 0x10)
-        {
-            return VER_TIME;
-        }
-        else if ((val & 0xF0) == 0x20)
-        {
-            return VER_DCE;
-        }
-        else if ((val & 0xF0) == 0x30)
-        {
-            return VER_NAME_MD5;
-        }
-        else if ((val & 0xF0) == 0x40)
-        {
-            return VER_RANDOM;
-        }
-        else if ((val & 0xF0) == 0x50)
-        {
-            return VER_NAME_SHA1;
-        }
-        else
-        {
-            return VER_UNKNOWN;
-        }
+        memcpy(m_data, &guid, 16);
+        // make big endian (internal storage type), windows is little endian
+        AZStd::endian_swap(*reinterpret_cast<AZ::u32*>(&m_data[0]));
+        AZStd::endian_swap(*reinterpret_cast<AZ::u16*>(&m_data[4]));
+        AZStd::endian_swap(*reinterpret_cast<AZ::u16*>(&m_data[6]));
     }
 
-    //=========================================================================
-    // operator<
-    // [4/11/2012]
-    //=========================================================================
-    bool Uuid::operator<(const Uuid& rhs) const
+    Uuid::operator GUID() const
     {
-        return AZStd::lexicographical_compare(data, data + AZ_ARRAY_SIZE(data), rhs.data, rhs.data + AZ_ARRAY_SIZE(rhs.data));
+        GUID guid;
+        memcpy(&guid, m_data, 16);
+        // make big endian (internal storage type), windows is little endian
+        AZStd::endian_swap(guid.Data1);
+        AZStd::endian_swap(guid.Data2);
+        AZStd::endian_swap(guid.Data3);
+        return guid;
     }
 
-    //=========================================================================
-    // operator>
-    // [4/11/2012]
-    //=========================================================================
-    bool Uuid::operator>(const Uuid& rhs) const
+    Uuid& Uuid::operator=(const GUID& guid)
     {
-        return AZStd::lexicographical_compare(rhs.data, rhs.data + AZ_ARRAY_SIZE(rhs.data), data, data + AZ_ARRAY_SIZE(data));
+        return operator=(Uuid(guid));
     }
-
-    //=========================================================================
-    // operator+
-    //=========================================================================
-    Uuid Uuid::operator + (const Uuid& rhs) const
+    bool Uuid::operator==(const GUID& guid) const
     {
-        u8 mergedData[sizeof(data) * 2];
-        memcpy(mergedData, data, sizeof(data));
-        memcpy(mergedData + sizeof(data), rhs.data, sizeof(data));
-        return CreateData(&mergedData, AZ_ARRAY_SIZE(mergedData));
+        return operator==(Uuid(guid));
     }
-
-    Uuid::FixedString Uuid::ToFixedString(bool isBrackets, bool isDashes) const
+    bool Uuid::operator!=(const GUID& guid) const
     {
-        return ToString<FixedString>(isBrackets, isDashes);
+        return !operator==(guid);
```

Comment:
Maybe all of these operators taking `GUID` should do the conversion right away: `return !operator==Uuid(guid);`.
(Repeat below)

Reply:
Actually I'll look into if we can remove the GUID functions in general.